functional-tests/functional/input/src/cpp/MultipleInheritance.cpp

```diff
@@ -77,4 +80,47 @@ MultipleInheritanceFactory::get_multi_interface() {
     return std::make_shared<MultiInterfaceImpl>();
 }
 
+std::shared_ptr<NarrowInterface>
+MultipleInheritanceFactory::get_multi_class_as_narrow() {
+    return std::make_shared<MultiClassImpl>();
+}
+
+std::shared_ptr<NarrowInterface>
+MultipleInheritanceFactory::get_multi_class_singleton() {
+    return s_multi_class;
+}
+
+std::shared_ptr<NarrowInterface>
+MultipleInheritanceFactory::upcast_multi_interface_to_narrow(const std::shared_ptr<MultiInterface>& instance) {
+    return std::dynamic_pointer_cast<NarrowInterface>(instance);
```

Comment:
Same as in Java, should work implicitly  here and below

Reply:
Done